core/src/main/scala/org/apache/spark/util/ListenerBus.scala

```diff
@@ -111,7 +118,10 @@ private[spark] trait ListenerBus[L <: AnyRef, E] extends Logging {
           logError(s"Listener ${Utils.getFormattedClassName(listener)} threw an exception", e)
       } finally {
         if (maybeTimerContext != null) {
```

Comment:
Maybe better to also guard with try-catch to not throwing NonFatal in finally statement, as throwing exception here may crash the dispatcher thread.